iTorrent/ViewControllers/TorrentDetails/TrackersList/TrackersListController.swift

```diff
@@ -112,23 +112,147 @@ class TrackersListController: ThemedUIViewController {
+                // do post add tasks
+                Dialog.show(self, title: title, message: "\(message)\n\(stats)")
+                postAddActions?()
+            }
+        }
+    }
+    
+    private func isValidTrackerURL(_ url: String) -> Bool {
+        // Regular expression pattern to validate a tracker URL
+        let scheme              = #"https?|udp|ftp|torrent|magnet|ws|wss"#
+        let port                = #":[0-9]+"#
+        let ipv4_addr_domain    = #"[0-9A-Za-z.-]+"#                       // ipv4 addr or domain names
+        let ipv6_addr           = #"\[?[0-9A-Fa-f:]+\]?"#                  // ipv6 addr
+        let path                = #"(/[A-Za-z0-9.-]+)*"#
+        
+        let pattern = "^(\(scheme))://(\(ipv4_addr_domain)|\(ipv6_addr))(\(port))?\(path)?$"
+        
+        if let regex = try? NSRegularExpression(pattern: pattern, options: .caseInsensitive) {
+            let range = NSRange(location: 0, length: url.utf16.count)
+            if let _ = regex.firstMatch(in: url, options: [], range: range) {
+                if let _ = URL(string: url) {
+                    return true
+                }
             }
         }
+        
+        return false
+    }
+
+    private func parseTorrentTrackersList(_ input: String,
+                                          _ pvc: ProgressViewController,
+                                          allowedProgress: Float = 100.0
+    ) -> (validTrackers: [String], entries: [String], lines: Int) {
+        // Split the input string into lines, removing empty lines and trimming spaces
+        let lines: [String] = input.components(separatedBy: .newlines)
+        let entries = lines
+            .map { $0.trimmingCharacters(in: .whitespaces) }
+            .filter { !$0.isEmpty }
+        
+        // update progress status
+        let consumed:Float = 5
+        let remaining:Float = allowedProgress - consumed
+        pvc.setProgress(pvc.getProgress()+consumed)         // bump progress by 5 percentage
+        
+        let increment = (remaining/Float(entries.count))
+        // Filter out the valid tracker URLs (remove duplicates if any by passing through set)
+        let validTrackers = Array(Set(entries.filter {
+            let validity = isValidTrackerURL($0)
+            pvc.setProgress(pvc.getProgress()+increment)    // update the progress
+            return validity
+        }))
+        print("parseTorrentTrackersList() progress: \(pvc.getProgress())")
+        return (validTrackers, entries, lines.count)
     }
 
     @IBAction func removeAction(_ sender: UIBarButtonItem) {
-        let controller = ThemedUIAlertController(title: nil, message: NSLocalizedString("Are you shure to remove this trackers?", comment: ""), preferredStyle: .actionSheet)
+        let controller = ThemedUIAlertController(title: nil, message: NSLocalizedString("Are you sure to remove this trackers?", comment: ""), preferredStyle: .actionSheet)
```

Comment:
Could you please update localization keys as well? (Or just leave key as it is and just fix the localization itself) Otherwise it will break other localizations

Reply:
@XITRIX, I have made the required changes in new commit e9a4d3e

Thanks